src/components/startups/Welcome.vue

```diff
@@ -185,39 +185,34 @@ export default {
                 return;
             }
 
-            let net;
-            if (!this.network) {
-                let netAddress = _.trim(options.server);
-
-                // Check if we have this network already
-                net = state.getNetworkFromAddress(netAddress);
-
-                // If we retreived an existing network, update the nick+password with what
-                // the user has just put in place
-                if (net) {
-                    net.nick = this.nick;
-                    net.connection.password = this.password;
-                }
+            let netAddress = _.trim(options.server);
 
-                // If the network doesn't already exist, add a new one
-                net = net || state.addNetwork('Network', this.nick, {
-                    server: netAddress,
-                    port: options.port,
-                    tls: options.tls,
-                    password: this.password,
-                    encoding: _.trim(options.encoding),
-                    direct: !!options.direct,
-                    path: options.direct_path || '',
-                    gecos: options.gecos,
-                });
-
-                if (options.recaptchaSiteId) {
-                    net.captchaResponse = this.captchaResponse();
-                }
-                this.network = net;
-            } else {
-                net = this.network;
+            // Check if we have this network already
+            let net = this.network || state.getNetworkFromAddress(netAddress);
+
+            // If we retreived an existing network, update the nick+password with what
+            // the user has just put in place
+            if (net) {
+                net.nick = this.nick;
+                net.connection.password = this.password;
+            }
+
+            // If the network doesn't already exist, add a new one
+            net = net || state.addNetwork('Network', this.nick, {
+                server: netAddress,
+                port: options.port,
+                tls: options.tls,
+                password: this.password,
+                encoding: _.trim(options.encoding),
+                direct: !!options.direct,
+                path: options.direct_path || '',
+                gecos: options.gecos,
+            });
+
+            if (!this.network && options.recaptchaSiteId) {
```

Comment:
I added !this.network so it would only ask the first time as that's what it was originally doing, not sure if this is correct due to not having recaptcha setup.

Does recaptcha time out? would a second connection attempt work with the first recaptacha answer?